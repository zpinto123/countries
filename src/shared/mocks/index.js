export const countries = [
  {
    code: "AD",
    name: "Andorra",
    currency: "EUR",
    phone: "376",
    continent: {
      name: "Europe"
    },
    languages: [
      {
        name: "Catalan",
        native: "Català"
      }
    ]
  },
  {
    code: "AE",
    name: "United Arab Emirates",
    currency: "AED",
    phone: "971",
    continent: {
      name: "Asia"
    },
    languages: [
      {
        name: "Arabic",
        native: "العربية"
      }
    ]
  },
  {
    code: "AF",
    name: "Afghanistan",
    currency: "AFN",
    phone: "93",
    continent: {
      name: "Asia"
    },
    languages: [
      {
        name: "Pashto",
        native: "پښتو"
      },
      {
        name: "Uzbek",
        native: "Ўзбек"
      },
      {
        name: "Turkmen",
        native: "Туркмен / تركمن"
      }
    ]
  }
];
