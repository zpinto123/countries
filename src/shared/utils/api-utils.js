const makeQuery = ({ client, query }) =>
  client
    .query({ query })
    .then(verifyResponse)
    .catch(handleErrors);

const verifyResponse = res => {
  if (!res || !res.data) return null;
  return res;
};

const handleErrors = error => error;

export { makeQuery };
