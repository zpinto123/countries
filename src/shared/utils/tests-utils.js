import configureStore from "redux-mock-store";
import countries from "../state/countries/initial-state";

const mockStore = configureStore();
const store = mockStore({ countries });

export { store };
