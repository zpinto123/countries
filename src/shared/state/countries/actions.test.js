import { store } from "../../utils/tests-utils";
import { countries as countriesMock } from "../../mocks";

const setCountries = countries => ({
  type: "SET_COUNTRIES",
  payload: { countries }
});

describe("action creators", () => {
  it("Saves countries to redux if getCountries is successful", () => {
    store.dispatch(setCountries(countriesMock));
    expect(store.getActions()).toMatchSnapshot();
  });
});
