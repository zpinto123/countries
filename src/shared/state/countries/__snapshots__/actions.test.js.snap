// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action creators Saves countries to redux if getCountries is successful 1`] = `
Array [
  Object {
    "payload": Object {
      "countries": Array [
        Object {
          "code": "AD",
          "continent": Object {
            "name": "Europe",
          },
          "currency": "EUR",
          "languages": Array [
            Object {
              "name": "Catalan",
              "native": "Català",
            },
          ],
          "name": "Andorra",
          "phone": "376",
        },
        Object {
          "code": "AE",
          "continent": Object {
            "name": "Asia",
          },
          "currency": "AED",
          "languages": Array [
            Object {
              "name": "Arabic",
              "native": "العربية",
            },
          ],
          "name": "United Arab Emirates",
          "phone": "971",
        },
        Object {
          "code": "AF",
          "continent": Object {
            "name": "Asia",
          },
          "currency": "AFN",
          "languages": Array [
            Object {
              "name": "Pashto",
              "native": "پښتو",
            },
            Object {
              "name": "Uzbek",
              "native": "Ўзбек",
            },
            Object {
              "name": "Turkmen",
              "native": "Туркмен / تركمن",
            },
          ],
          "name": "Afghanistan",
          "phone": "93",
        },
      ],
    },
    "type": "SET_COUNTRIES",
  },
]
`;
